src/common/dns_utils.cpp

```diff
@@ -447,19 +447,28 @@ bool load_txt_records_from_dns(std::vector<std::string> &good_records, const std
   std::uniform_int_distribution<int> dis(0, dns_urls.size() - 1);
   size_t first_index = dis(gen);
 
-  bool avail, valid;
+  // send all requests in parallel
+  std::vector<boost::thread> threads(dns_urls.size());
+  std::deque<bool> avail(dns_urls.size(), false), valid(dns_urls.size(), false);
+  for (size_t n = 0; n < dns_urls.size(); ++n)
+  {
+    threads[n] = boost::thread([n, dns_urls, &records, &avail, &valid](){
+      records[n] = tools::DNSResolver::instance().get_txt_record(dns_urls[n], avail[n], valid[n]); 
+    });
+  }
+  for (size_t n = 0; n < dns_urls.size(); ++n)
+    threads[n].join();
```

Comment:
FWIW, this is:
```c++
std::vector<boost::future<std::vector<std::string>>> records;
records.push_back(
    boost::async([&, n] () {
        return tools::DNSResolver::instance().get_tx_record(dns_urls[n], avail[n], valid[n])
    }
);

```
Which blocks until the thread joins if a `boost::thread` constructor throws after a previous one was created successfully. This is unlikely to happen, but if you like the syntax better it does prevent that rare situation.

Reply:
OK. I use simple code, and am loathe to change it on review to code that may be better, but which I do not understand well. You're welcome to change it later to this if you think it's better though.